src/api/api.ts

```diff
@@ -0,0 +1,251 @@
+/* eslint-disable @typescript-eslint/no-throw-literal */
+import axios from 'axios';
+
+import axiosInstance from './axiosInstance';
+
+// api 타입 정의
+export interface GetRankingProductsRequest {
+  targetType?: 'ALL' | 'FEMALE' | 'MALE' | 'TEEN';
+  rankType?: 'MANY_WISH' | 'MANY_RECEIVE' | 'MANY_WISH_RECEIVE';
+}
+
+export interface GetRankingProductsResponse {
+  products: ProductData[];
+}
+
+export interface GetThemesResponse {
+  themes: ThemeData[];
+}
+
+export interface GetThemeProductsRequest {
+  themeKey: string;
+  pageToken?: string;
+  maxResults?: number;
+}
+
+export interface GetThemeProductsResponse {
+  products: ProductData[];
+  nextPageToken?: string;
```

Comment:
```suggestion
  nextPageToken?: string | null;
```

oas에 `nullable: true`로 표현되어 있지는 않긴 한데 description에 `만약 없다면 null`로 적혀있어서 nullable 표시해주는게 좋겠습니다.

Reply:
step3 브랜치에서 반영하였습니다
<img width="372" alt="image" src="https://github.com/user-attachments/assets/caa5aa8e-714d-4589-a7a0-b1c890817c89">